homeassistant/components/xiaomi_miio/humidifier.py

```diff
@@ -0,0 +1,660 @@
+)
+from miio.airhumidifier_miot import (  # pylint: disable=import-error, import-error
+    LedBrightness as AirhumidifierMiotLedBrightness,
+    OperationMode as AirhumidifierMiotOperationMode,
+    PressedButton as AirhumidifierPressedButton,
+)
+import voluptuous as vol
+
+from homeassistant.components.humidifier import PLATFORM_SCHEMA, HumidifierEntity
+from homeassistant.components.humidifier.const import (
+    DEVICE_CLASS_HUMIDIFIER,
+    SUPPORT_MODES,
+)
+from homeassistant.const import ATTR_ENTITY_ID, CONF_HOST, CONF_NAME, CONF_TOKEN
+import homeassistant.helpers.config_validation as cv
+
+from .const import (
+    CONF_DEVICE,
+    CONF_FLOW_TYPE,
+    DOMAIN,
+    MODEL_AIRHUMIDIFIER_CA1,
+    MODEL_AIRHUMIDIFIER_CA4,
+    MODEL_AIRHUMIDIFIER_CB1,
+    MODELS_HUMIDIFIER,
+    MODELS_HUMIDIFIER_MIOT,
+    SERVICE_HUMIDIFIER_SET_BUZZER_OFF,
+    SERVICE_HUMIDIFIER_SET_BUZZER_ON,
+    SERVICE_HUMIDIFIER_SET_CHILD_LOCK_OFF,
+    SERVICE_HUMIDIFIER_SET_CHILD_LOCK_ON,
+    SERVICE_HUMIDIFIER_SET_DRY_OFF,
+    SERVICE_HUMIDIFIER_SET_DRY_ON,
+    SERVICE_HUMIDIFIER_SET_LED_BRIGHTNESS,
+    SERVICE_HUMIDIFIER_SET_LED_OFF,
+    SERVICE_HUMIDIFIER_SET_LED_ON,
+    SERVICE_HUMIDIFIER_SET_MOTOR_SPEED,
+    SERVICE_HUMIDIFIER_SET_TARGET_HUMIDITY,
+)
+from .device import XiaomiMiioEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = "Xiaomi Miio Device"
+DEFAULT_MIN_HUMIDITY = 30
+DEFAULT_MAX_HUMIDITY = 80
+DATA_KEY = "humidifier.xiaomi_miio"
+
+CONF_MODEL = "model"
+
+ATTR_MODEL = "model"
+
+MODE_AUTO = "Auto"
+MODE_HIGH = "High"
+MODE_LOW = "Low"
+MODE_MEDIUM = "Medium"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_HOST): cv.string,
+        vol.Required(CONF_TOKEN): vol.All(cv.string, vol.Length(min=32, max=32)),
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_MODEL): vol.In(MODELS_HUMIDIFIER),
+    }
+)
+
+ATTR_TEMPERATURE = "temperature"
+ATTR_HUMIDITY = "humidity"
+ATTR_BUZZER = "buzzer"
+ATTR_CHILD_LOCK = "child_lock"
+ATTR_LED = "led"
+ATTR_LED_BRIGHTNESS = "led_brightness"
+ATTR_BRIGHTNESS = "brightness"
+ATTR_LEVEL = "level"
+ATTR_CURRENT_HUMIDITY = "current_humidity"
+ATTR_FEATURES = "features"
+ATTR_VOLUME = "volume"
+ATTR_USE_TIME = "use_time"
+ATTR_BUTTON_PRESSED = "button_pressed"
+
+# Air Humidifier
+ATTR_TARGET_HUMIDITY = "target_humidity"
+ATTR_TRANS_LEVEL = "trans_level"
+ATTR_HARDWARE_VERSION = "hardware_version"
+
+# Air Humidifier CA
+ATTR_MOTOR_SPEED = "motor_speed"
+ATTR_DEPTH = "depth"
+ATTR_DRY = "dry"
+
+# Air Humidifier CA4
+ATTR_ACTUAL_MOTOR_SPEED = "actual_speed"
+ATTR_FAHRENHEIT = "fahrenheit"
+ATTR_FAULT = "fault"
+ATTR_WATER_LEVEL = "water_level"
+
+AVAILABLE_ATTRIBUTES_AIRHUMIDIFIER_COMMON = {
+    ATTR_TEMPERATURE: "temperature",
+    ATTR_CURRENT_HUMIDITY: "humidity",
+    ATTR_BUZZER: "buzzer",
+    ATTR_CHILD_LOCK: "child_lock",
+    ATTR_LED_BRIGHTNESS: "led_brightness",
+    ATTR_USE_TIME: "use_time",
```

Comment:
Times are best exposed as sensor entities with timestamp device class. What kind of time is this?

Reply:
It's an increasing number of "powered-on" hours.

Reply:
Thanks! That should then be converted to an absolute isoformatted UTC datetime string and reported in a separate sensor.

https://developers.home-assistant.io/docs/core/entity/sensor#available-device-classes